### PR TITLE
Refactor scheduler tests: split out prompt generation

### PR DESCRIPTION
**Pull request by joerunde**
# Description

Refactors for tkv step tests. This splits out the prompt creation from the engine validation, so that we can be very explicit about what prompts we are constructing for the tests.

IMHO, it makes the tests for these prefix edge cases more understandable, though it is a bit more verbose for the simple case where the prompts are just identical.

### CODE REVIEW

File: tests/scheduling_utils.py [hunk 1]
```diff
@@ -38,6 +40,92 @@ def augment_checked_steps(
...
+        seed=seed,
+        num_tokens=length
+    ).prompt_token_ids
+
+def create_request_for_scheduler_test(
+    model: ModelInfo,
+    request_id: int,
+    add_step: int,
+    max_tokens: int,
+    prompt: list[int],
```
[the remark this answers is not in the document]
Comment (prashantgupta24) [reply tests/scheduling_utils.py]: oh is that why you created the `random_prompt` function?

File: tests/e2e/test_spyre_pc_scheduler_steps.py [hunk 2]
```diff
@@ -289,24 +289,36 @@ def test_prefix_hit_decoded_block_within_batch(
...
         monkeypatch=monkeypatch,
-        seqs_max_tokens=seqs_max_tokens,
-        prompts_lengths=prompts_lengths,
-        steps_add_reqs=steps_add_reqs,
+        requests=[request1, request2],
         checked_steps=checked_steps,
         max_num_seqs=max_num_seqs,
         max_model_len=max_model_len,
         available_blocks=available_blocks,
+        max_batch_tkv_limit=-1,
```
Comment (prashantgupta24) [inline tests/e2e/test_spyre_pc_scheduler_steps.py]: not sure about this

File: tests/e2e/test_spyre_pc_scheduler_steps.py [hunk 3]
```diff
@@ -1161,14 +1229,27 @@ def test_multi_chunk_partial_match(model: ModelInfo, backend: str,
...
+        use_golden_token_injection=True)
+
+    request2 = create_request_for_scheduler_test(
+        model=model,
+        request_id=1,
+        add_step=0,
+        max_tokens=2,
+        prompt=request1.request.prompt_token_ids[0:254] +
+        random_prompt(model=model, seed=0, length=384 - 254),
+        use_golden_token_injection=True)
```
Comment (sducouedic) [inline tests/e2e/test_spyre_pc_scheduler_steps.py]: oh this is makes things actually so much simpler, no need to have the dedicated logic to create partial prefix prompt in the functions, just for a simple test. 

But I would put prompt 2 in a dedicated variable, for readability and highlighting that it is partial prefix prompt

```suggestion
    prompt2= request1.request.prompt_token_ids[0:254] +
        random_prompt(model=model, seed=0, length=384 - 254)
    request2 = create_request_for_scheduler_test(
        model=model,
        request_id=1,
        add_step=0,
        max_tokens=2,
        prompt=prompt2,
        use_golden_token_injection=True)
```

File: tests/e2e/test_spyre_pc_scheduler_steps.py [hunk 4]
```diff
@@ -177,15 +183,24 @@ def test_prefix_hit_decoded_block_within_batch(
...
+        prompt=random_prompt(model=model, seed=0, length=126),
+        use_golden_token_injection=True)
+
+    request2 = create_request_for_scheduler_test(
+        model=model,
+        request_id=1,
+        add_step=67,
+        max_tokens=2,
+        prompt=request1.request.prompt_token_ids +
+        request1.hf_output["token_ids"][:2] +
```
Comment (sducouedic) [inline tests/e2e/test_spyre_pc_scheduler_steps.py]: here also I would put in `prompt2` variable

File: tests/e2e/test_spyre_pc_scheduler_steps.py (hunk 3 above)
Comment (joerunde) [reply tests/e2e/test_spyre_pc_scheduler_steps.py]: Thanks @sducouedic! 

I went ahead and did this for all the test cases, they're all more readable now 🚀🚀🚀

File: tests/scheduling_utils.py [hunk 5]
```diff
@@ -39,14 +42,81 @@ def augment_checked_steps(
...
+    # Can add golden token injection, which will ensure that the vllm output
+    # matches the hf output so that we can do logits comparisons on identical
+    # token outputs.
+
+    sampling_params = SamplingParams(max_tokens=max_tokens,
+                                     temperature=0.0,
+                                     logprobs=0,
+                                     ignore_eos=True)
+
+    hf_results = generate_hf_output(
```
Comment (yannicks1) [inline tests/scheduling_utils.py]: very nice how this is afterwards used in the hf vllm comparison!

**Review by yannicks1 — APPROVED** [review]
This is amazing work! much more readable now. Thanks a lot @joerunde and @prashantgupta24 !
